external/webserver/http_client.c

```diff
@@ -476,6 +477,30 @@ int http_parse_message(char *buf, int buf_len, int *method, char *url,
 						http_dispatch_url(client, req);
 						process_finish = true;
 						read_finish = true;
+
+						/* Process the Trailer */
+						header_field_value = http_keyvalue_list_find(params, "Trailer");
+						if (strncmp(header_field_value, "(null)", 6)) {
+							len->sentence_start = sentence_end + 2;
+
+							/* match the Trailer field */
+							if (strncmp(buf + len->sentence_start, header_field_value, strlen(header_field_value)) == 0) {
```

Comment:
compare length strlen doesn't check null character. So you need to modify it.

Reply:
Then how about to check '\r' if it's end of line? It's more safe

Reply:
I have added the additional validation of collon (:) after trailer header (Expires)